plugins/optimization-detective/tests/test-optimization.php

```diff
@@ -69,6 +71,44 @@ function ( $buffer ) use ( $original, $expected ) {
 
 		$buffer = ob_get_clean(); // Get the buffer from our wrapper output buffer.
 		$this->assertSame( $expected, $buffer );
+		$this->assertTrue( $filter_invoked );
+	}
+
+	/**
+	 * Make output is buffered and that it is also filtered.
+	 *
+	 * @covers ::od_buffer_output
+	 */
+	public function test_od_buffer_output_not_finalized(): void {
+		$original = 'Hello My World!';
+		$override = 'Ciao mondo!';
+
+		// In order to test, a wrapping output buffer is required because ob_get_clean() does not invoke the output
+		// buffer callback. See <https://stackoverflow.com/a/61439514/93579>.
+		ob_start();
+
+		$filter_invoked = false;
+		add_filter(
+			'od_template_output_buffer',
+			function ( $buffer ) use ( $original, &$filter_invoked ) {
+				$this->assertSame( $original, $buffer );
```

Comment:
This assertion fails. `$buffer` is unexpectedly `Ciao mondo!` when it is expected to be an empty string. Also, the callback is being called, so `ob_clean()` is clearly not passing a non-`PHP_OUTPUT_HANDLER_FINAL` flag as the phase.